crates/flux-refineck/src/checker.rs

```diff
@@ -1398,7 +1400,10 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
                 let idx = Expr::constant(rty::Constant::from(*s));
                 Ok(Ty::mk_ref(ReStatic, Ty::indexed(BaseTy::Str, idx), Mutability::Not))
             }
-            Constant::Char => Ok(Ty::char()),
+            Constant::Char(c) => {
+                let idx = Expr::constant(rty::Constant::from(*c as u128));
```

Comment:
```suggestion
                let idx = Expr::constant(rty::Constant::Char(c));
```
or `rty::Constant::from(c)` if you created the impl